src/Annotation.js

```diff
@@ -206,9 +206,27 @@ Potree.Annotation = class extends THREE.EventDispatcher {
 		}
 	}
 
+	hasChild(annotation) {
+		return this.children.includes(annotation);
+	}
+
 	remove (annotation) {
-		this.children = this.children.filter(e => e !== annotation);
-		annotation.parent = null;
+		if (this.hasChild(annotation)) {
```

Comment:
Won't this always evaluate to false, since you've removed the child from this.children two lines before?